spock-core/src/main/java/org/spockframework/runtime/GroovyRuntimeUtil.java

```diff
@@ -301,4 +302,16 @@ public static Method toMethod(@Nullable MetaMethod metaMethod) {
     CachedMethod cachedMethod = ObjectUtil.asInstance(metaMethod, CachedMethod.class);
     return cachedMethod == null ? null : cachedMethod.getCachedMethod();
   }
+
+  @Internal
+  public static boolean isGroovy2() {
+    return GroovySystem.getVersion().startsWith("2.");
+  }
+
+  @Internal
+  public static boolean isGroovy3() {
+    //In fact it is "isNotGroovy2()", but "isGroovy3()" sounds better and before Groovy 4 is on the horizon it leads to the same
+    //With "startsWith("3.") there could be (in the future) no tests executed at all for Groovy 4
+    return !isGroovy2();
```

Comment:
Why not `return GroovySystem.getVersion().startsWith("3.");` ?

Reply:
With `startsWith("3.")` there could be (in the future) no tests executed at all for Groovy 4.

Reply:
Then maybe it should be called `isGroovy3OrNewer` or `isAtLeatGroovy3` or similar, or do not use `isGroovy3()` but `isGroovy2()`?

Reply:
You are right that `isGroovy3OrNewer()` is more accurate. I updated the code.

Due to `3Or` it doesn't look as good as `isAtLeastGroovy3()`, but `isGroovy3OrNewer()` is more consistent with `isGroovy2()`

Reply:
Yep, it just reads strange, especially if you only look at the three chars you posted.
Just reading your comment I read it as "thirty-r" and wondered what you mean. :-D
Because of that I provided a more readable alternative.
Maybe a middle-way: `isGroovy3orNewer`?
It doesn't strictly follow camel-case naming conventions, but might be a bit easier to read as the lower o cannot as easily confused with the 0 as the upper O.

Reply:
In the end there is `isGroovy3orNewer`. Thanks @Vampire for your idea.